test/preview/components/retrievers/test_memory_retriever.py

```diff
@@ -0,0 +1,125 @@
+from typing import Dict, Any, List
+
+import pytest
+
+from haystack.preview import Pipeline
+from haystack.preview.components.retrievers.memory import MemoryRetriever
+from haystack.preview.dataclasses import Document
+from haystack.preview.document_stores import MemoryDocumentStore
+
+from test.preview.components.base import BaseTestComponent
+
+
+@pytest.fixture()
+def test_memory_retriever_docs():
+    return [
+        Document.from_dict({"content": "Javascript is a popular programming language"}),
+        Document.from_dict({"content": "Java is a popular programming language"}),
+        Document.from_dict({"content": "Python is a popular programming language"}),
+        Document.from_dict({"content": "Ruby is a popular programming language"}),
+        Document.from_dict({"content": "PHP is a popular programming language"}),
+    ]
+
+
+class Test_MemoryRetriever(BaseTestComponent):
+    @pytest.mark.unit
+    def test_save_load(self, tmp_path):
+        self.assert_can_be_saved_and_loaded_in_pipeline(MemoryRetriever(document_store_name="memory"), tmp_path)
+
+    @pytest.mark.unit
+    def test_save_load_with_parameters(self, tmp_path):
+        self.assert_can_be_saved_and_loaded_in_pipeline(
+            MemoryRetriever(document_store_name="memory", top_k=5, scale_score=False), tmp_path
+        )
+
+    @pytest.mark.unit
+    def test_init_default(self):
+        retriever = MemoryRetriever(document_store_name="memory")
+        assert retriever.document_store_name == "memory"
+        assert retriever.defaults == {"top_k": 10, "scale_score": True}
+
+    @pytest.mark.unit
+    def test_init_with_parameters(self):
+        retriever = MemoryRetriever(document_store_name="memory-test", top_k=5, scale_score=False)
+        assert retriever.document_store_name == "memory-test"
+        assert retriever.defaults == {"top_k": 5, "scale_score": False}
+
+    @pytest.mark.unit
+    def test_init_with_invalid_top_k_parameter(self):
+        with pytest.raises(ValueError, match="top_k must be > 0, but got -2"):
+            MemoryRetriever(document_store_name="memory-test", top_k=-2, scale_score=False)
+
+    @pytest.mark.unit
+    def test_valid_run(self, test_memory_retriever_docs):
+        ds = MemoryDocumentStore()
+        ds.write_documents(test_memory_retriever_docs)
+        mr = MemoryRetriever(document_store_name="memory")
+        result: MemoryRetriever.Output = mr.run(data=MemoryRetriever.Input(query="PHP", stores={"memory": ds}))
+
+        assert result.documents
+        assert len(result.documents) <= len(test_memory_retriever_docs)
```

Comment:
You can make this check exact: 
```suggestion
        assert len(result.documents) == mr.top_k
```